src/goto-checker/properties.h

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************\
+
+Module: Properties
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Properties
+
+#ifndef CPROVER_GOTO_CHECKER_PROPERTIES_H
+#define CPROVER_GOTO_CHECKER_PROPERTIES_H
+
+#include <unordered_map>
+
+#include <goto-programs/goto_model.h>
+
+/// The result status of a property
+enum class property_resultt
+{
+  /// The property was not reached (also used for initialization)
+  NOT_REACHED,
+  /// The status of the property is unknown
+  UNKNOWN,
+  /// The property was proven to be unreachable
+  NOT_REACHABLE,
+  /// The property was not violated
+  PASS,
+  /// The property was violated
+  FAIL,
+  /// An error occurred during goto checking
+  ERROR
+};
+
+std::string as_string(property_resultt result);
+
+/// The result of goto checking and verifying
+enum class resultt
+{
+  /// No property was violated, neither was there an error
+  UNKNOWN,
+  /// No properties were violated
+  PASS,
+  /// Some properties were violated
+  FAIL,
+  /// An error occurred during goto checking
+  ERROR
+};
+
+std::string as_string(resultt result);
+
+struct property_infot
+{
+  /// A pointer to the corresponding goto instruction
+  goto_programt::const_targett location;
```

Comment:
The naming is a bit confusing as I would expect `location` to be a `source_locationt`. Could it be renamed to something like `instruction_pointer`?

Reply:
Going with `pc` which is used in other classes such as `goto_tracet`.